notebooks/query-language-object-based-dialect/01.01.quantifier_capabilities.ipynb

```diff
@@ -0,0 +1,842 @@
+    "\n",
+    "\n",
+    "A hatchet query always finds all **matching paths** for a provided GraphFrame. For this reason, Hatchet queries also takes the shape of a sequence of **predicates** to match, with the added convienience of also allowing to define an accompinying **quantifier** for each predicate. This concept is reflected in Fig. 2, where we interpret queries in terms of filtering or selecting parts of the GraphFrame."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f79bfc24",
+   "metadata": {},
+   "source": [
+    "![Graph frames and queries](../common/images/hatchet_query_graphframe.png)\n",
+    "\n",
+    "**Figure 2**: A diagram to provide an overview of queries and an example of how queries filter GraphFrames."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "51ee5d2b",
+   "metadata": {},
+   "source": [
+    "***"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "741f0361",
+   "metadata": {},
+   "source": [
+    "## Category 1: Quantifier Capabilities\n",
+    "\n",
+    "A valid hatchet query requires a **quantifier**. The accepted values for query node quantifiers in the **Object-based Dialect** are:\n",
+    "\n",
+    "1. `\".\"`: Match a single node\n",
+    "2. `\"*\"`: Match 0 or more nodes\n",
+    "3. `\"+\"`: Match 1 or more nodes\n",
+    "4. `Integer`: Match an exact number of nodes"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fb2db956",
+   "metadata": {},
+   "source": [
+    "### Loading profile data as Hatchet GraphFrame\n",
+    "\n",
+    "Hatchet queries are only defined on Hatchet GraphFrames. \n",
+    "Obtaining a hatchet GraphFrame is straight forward:\n",
+    "\n",
+    "1. Import hatchet\n",
+    "2. Use the appropriete reader for the profile/trace at hand\n",
+    "\n",
+    "This notebook loads a [Caliper](https://github.com/LLNL/Caliper) profile in JSON format and Caliper is a performance profiling library.\n",
+    "\n",
+    "The dataset contains [LULESH (Livermore Unstructured Lagrangian Explicit Shock Hydrodynamics)](https://asc.llnl.gov/codes/proxy-apps/lulesh) performance report data generated by Caliper. LULESH is a highly simplified application designed to solve the Sedov Blast problem, which is a standard hydrodynamics test problem. It performs a hydronomics stencil calculation using both MPI and OpenMP to achieve parallelism. \n",
+    "\n",
+    "This is an interesting profile because it covers a relatively large number of nodes and also spends considerable time in MPI communication routines."
```

Comment:
This is an interesting example because it comprises a large number of nodes [specify number here maybe?] and most of the runtime comes from the MPI communication routines.